ertviz/controllers/response_correlation_controller.py

```diff
@@ -0,0 +1,273 @@
+                    y_axis=data_df[x_val],
+                    text="Mean",
+                    name=f"Mean {repr(ensemble)}",
+                    **style,
+                )
+                for x_val in data_df
+            ]
+
+            for obs in response.observations:
+                _plots += _get_observation_plots(obs.data_df(), x_axis)
+        fig = go.Figure()
+        for plot in _plots:
+            fig.add_trace(plot.repr)
+
+        _layout = assets.ERTSTYLE["figure"]["layout"].copy()
+        _layout.update(dict(showlegend=False))
+        fig.update_layout(_layout)
+        fig.add_shape(
+            type="line",
+            x0=x_axis[corr_xindex],
+            y0=0,
+            x1=x_axis[corr_xindex],
+            y1=1,
+            yref="paper",
+            line=dict(color="rgb(30, 30, 30)", dash="dash", width=3),
+        )
+        fig.update_layout(clickmode="event+select")
+        return fig
+
+    @app.callback(
+        Output(
+            {"id": parent.uuid("response-scatterplot"), "type": parent.uuid("graph")},
+            "figure",
+        ),
+        [
+            Input(parent.uuid("correlation-store-xindex"), "modified_timestamp"),
+            Input(parent.uuid("correlation-store-parameter"), "modified_timestamp"),
+            Input(parent.uuid("parameter-selection-store-resp"), "modified_timestamp"),
+            Input(parent.uuid("parameter-selection-store-param"), "modified_timestamp"),
+            Input(parent.uuid("ensemble-selection-store"), "modified_timestamp"),
+        ],
+        [
+            State(parent.uuid("parameter-selection-store-param"), "data"),
+            State(parent.uuid("parameter-selection-store-resp"), "data"),
+            State(parent.uuid("ensemble-selection-store"), "data"),
+            State(parent.uuid("correlation-store-xindex"), "data"),
+            State(parent.uuid("correlation-store-parameter"), "data"),
+        ],
+    )
+    def update_response_parameter_scatterplot(
+        _,
+        __,
+        ___,
+        ____,
+        _____,
+        parameters,
+        responses,
+        ensembles,
+        corr_xindex,
+        corr_parameter,
+    ):
+
+        if not parameters or not ensembles or not responses:
+            raise PreventUpdate
+
+        parameter = corr_parameter if corr_parameter else parameters[0]
+        _plots = []
+        _resp_plots = {}
+        _param_plots = {}
+        _colors = {}
+        for ensemble_id, data in ensembles.items():
+            ensemble = load_ensemble(parent, ensemble_id)
+            # take the first selected response
+            y_data = ensemble.parameters[parameter].data_df()
+            response = ensemble.responses[responses[0]]
+            x_data = response.data_df().loc[corr_xindex]
+            style = deepcopy(assets.ERTSTYLE["response-plot"]["response-index"])
+            style["marker"]["color"] = data["color"]
+            _colors[str(ensemble)] = data["color"]
```

Comment:
Guess not a big deal , but you `str(ensemble)` at least 3 times inside this function - would it make sense assigning it to a variable? Or - make it a property on the ensemble obj itself making it easy extracting the string representation of it? I think this is used also in other functions as well !

Reply:
The computational overhead is not as big in case of `str(ensemble)`, so it's more readability; but can assign it to `str_ensemble` somewhere above if you think its' readable.